space-time.Rmd

```diff
@@ -155,53 +166,112 @@ if (file.exists("mi_raster.rds")) {
+
+```{r}
+library(USAboundaries)
+library(sf)
+c18 <- us_boundaries(as.Date("1820-01-01"))
+class(c18)
 ```
 
-### Area metadata
+The key column in this data set is the `geometry` variable, which specifies the spatial geometry for each of the US state borders as they stood in 1820. 
 
-Sometimes metadata is associated not with a point, but with an area. For example, we can create `mi_census` which provides census information about each county in MI:
+```{r}
+c18[1:10, c("full_name", "geometry")]
+```
+
+Each element in the `geometry` column is a multipolygon object and --- as the name suggests --- contains data specifying the vertices of one or more polygons that demark the border of a single US state. The data printed at the beginning of the output indicates that there is additional useful mapping information stored in the `geometry` column. Given data in this format, we can use `geom_sf()` and `coord_sf()` to draw a serviceable map without specifying any parameters or even explicitly declaring any aesthetics:
 
 ```{r}
-mi_census <- midwest %>%
-  tbl_df() %>%
-  filter(state == "MI") %>% 
-  mutate(county = tolower(county)) %>%
-  select(county, area, poptotal, percwhite, percblack)
-mi_census
+ggplot(c18) + 
+  geom_sf() + 
+  coord_sf()
 ```
 
-We can't map this data directly because it has no spatial component. Instead, we must first join it to the vector boundaries data. This is not particularly space efficient, but it makes it easy to see exactly what data is being plotted. Here I use `dplyr::left_join()` to combine the two datasets and create a choropleth map. \index{Choropleth}
+To understand what this code does, note that `geom_sf()` relies on a `geometry` aesthetic that is not used elsewhere in ggplot2. This aesthetic can be specified in one of three ways. In the simplest case (illustrated above) when the user does nothing, `geom_sf()` will attempt to map it to a column named `geometry`. However, if the `data` argument is an sf object then `geom_sf()` is able to automatically detect a column that specifies a simple features geometry (discussed below), and constructs the mapping using that column regardless of what it is named. The third possibility is that the user can specify the mapping manually in the usual way with `aes(geometry = my_column)`. 
+
+The second thing to comment on in this code is how the `coord_sf()` function sets the coordinates for the plot. The geometry column is usually associated with a coordinate reference system (CRS), stored internally as one of the attributes of the column. Because this information is usually available, the default behaviour of `coord_sf()` is to extract the CRS from the sf geometry column and use that to construct the coordinate system for the plot (see Chapter \@ref(coord)). 
```

Comment:
I wonder if it's better to document `coord_sf()` here, in order to keep all the spatial stuff together?